pontoon/sync/utils.py

```diff
@@ -63,24 +70,34 @@ def directory_contains_resources(directory_path, source_only=False):
     return False
 
 
-def locale_directory_path(checkout_path, locale_code):
+def locale_directory_path(checkout_path, locale_code, parent_directories=None):
     """
     Path to the directory where strings for the given locale are
     stored.
     """
     possible_paths = []
-    for root, dirnames, filenames in os.walk(checkout_path):
-        if locale_code in dirnames:
-            possible_paths.append(os.path.join(root, locale_code))
 
-        locale_variant = locale_code.replace('-', '_')
-        if locale_variant in dirnames:
-            possible_paths.append(os.path.join(root, locale_variant))
+    # Check paths that use underscore as locale/country code separator
+    locale_variant = locale_code.replace('-', '_')
+
+    # Optimization for directories with a lot of paths
```

Comment:
I'd only add examples of directories that will be matched -> it will be a little easier to understand.